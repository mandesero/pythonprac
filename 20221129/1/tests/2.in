from itertools import accumulate


class Boba(metaclass=dump):
    def __init__(self, *args, **kwargs):
        print(args, kwargs)
        self.a = len(args)
        self.b = len(kwargs) + self.a

    def __iter__(self):
        return accumulate([self.a, self.b])


a = Boba(1, 2, 3, 4, 5, c=2, d=3)
print(sum(a))
