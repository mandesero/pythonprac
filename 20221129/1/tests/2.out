__init__: (1, 2, 3, 4, 5), {'c': 2, 'd': 3}
(1, 2, 3, 4, 5) {'c': 2, 'd': 3}
__iter__: (), {}
17
